tests/unit/kvdb/kvdb_ctxn_test.c

```diff
@@ -1485,4 +1487,49 @@ MTF_DEFINE_UTEST_PREPOST(kvdb_ctxn_test, multiple_ctxn_commit, mapi_pre, mapi_po
 }
 #endif
 
+MTF_DEFINE_UTEST_PREPOST(kvdb_ctxn_test, timeout, mapi_pre, mapi_post)
+{
+    merr_t err;
+    struct viewset *vs;
+    struct kvdb_ctxn *handle;
+    struct kvdb_ctxn_impl *ctxn;
+    struct c0snr_set *css;
+    struct kvdb_ctxn_set *set;
+    atomic_ulong kvdb_seq, tseqno;
+    enum kvdb_ctxn_state state;
+
+    atomic_set(&kvdb_seq, 1);
+    atomic_set(&tseqno, 0);
+
+    err = viewset_create(&vs, &kvdb_seq, &tseqno);
+    ASSERT_TRUE(err == 0);
+
+    /* 1 second timeout with a 1 second delay */
+    err = kvdb_ctxn_set_create(&set, 1000, 1000);
+    ASSERT_EQ(0, merr_errno(err));
+
+    err = c0snr_set_create(&css);
+    ASSERT_EQ(0, merr_errno(err));
+
+    handle = kvdb_ctxn_alloc(NULL, NULL, &kvdb_seq, set, vs, css, NULL, NULL);
+    ASSERT_NE(NULL, handle);
+
+    ctxn = kvdb_ctxn_h2r(handle);
+
+    err = kvdb_ctxn_begin(handle);
+    ASSERT_EQ(0, merr_errno(err));
+
+    /* Force the transaction to get internally aborted due to timeout */
+    sleep(2);
```

Comment:
This might end up being a non-deterministic test since it depends on the reaper executing while waiting here.

Reply:
I was running with `--repeat` and never saw it fail, but I agree with you on principle. Do you have any better ideas?

Reply:
The easiest better idea is to poll the state for a longer period of time and fail after, say 60 seconds (or even more) of waiting.  Normally it will succeed in a short time, so waiting for a long time won't penalize the normal success case.

Reply:
Makes sense. I'll do that.